tests/parametric/test_headers_precedence.py

```diff
@@ -39,6 +94,10 @@ class Test_Headers_Precedence:
     @missing_feature(context.library == "python", reason="New 'datadog' default hasn't been implemented yet")
     @missing_feature(context.library == "python_http", reason="New 'datadog' default hasn't been implemented yet")
     def test_headers_precedence_propagationstyle_default(self, test_agent, test_library):
+        self.test_headers_precedence_propagationstyle_datadog(test_agent, test_library)
```

Comment:
Doesn't this cause us to essentially just run `test_headers_precedence_propagationstyle_datadog` twice? Also, isn't @enable_datadog() just going to override the default, and the default should no longer be `datadog` propagation style? Seems like `test_headers_precedence_propagationstyle_default` should be renamed.

Reply:
Yes it runs the the code in `test_headers_precedence_propagationstyle_datadog` twice, since the previous default was `DD_PROPAGATION_STYLE=datadog`. Essentially the previous default or setting the configuration to the same value should yield the same results. I didn't want to rename this method because I think it makes sense for this to be the original default method, and we can add version ranges to this in the future to show when the default had changed.

And AFAIK the environment shouldn't be overriding when calling one method from another, unless my understanding of the pytest fixtures is incorrect.